### PR TITLE
Add packet broker UI

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
This PR adds Packet Broker related functionality to the Console.

#3771

<details><summary>Screenshots</summary>

image

image

image

image

</details>

#### Changes

- Add Packet Broker functionality to Console
  - Registering with Packet Broker
  - Setting default routing policy for the own network
  - Setting network specific routing policies
  - Examining networks
- Add `<RoutingPolicy.Sheet />`, `<RoutingPolicy.Matrix />` components
- Add `useBreadcrumbs()` hook
- Add `<Switch />` component
- Add routing policy form
- Add packet broker network table
- Update `<FetchTable />` to allow for more cell alignment options
- Add Packet Broker service to JS SDK
- Various related updates

#### Testing

Manual testing. Will work on e2e tests as well, but I think we can do that during the review already.

##### Regressions

None.

#### Notes for Reviewers

Packet broker needs to be configured in order to use the interface. @bafonins I'll send you instructions to configure the stack to connect to the staging server of packet broker.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/store/middleware/logics/packet-broker.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      dispatch(packetBroker.getHomeNetworkRoutingPoliciesSuccess(homeNetwork))
+    }
+
+    return { entities: data.networks, totalCount: data.totalCount }
+  },
+})
+
+const getPacketBrokerForwarderPoliciesLogic = createRequestLogic({
+  type: packetBroker.GET_FORWARDER_ROUTING_POLICIES,
+  process: async ({ action }) => {
```
Comment (bafonins) [inline pkg/webui/console/store/middleware/logics/packet-broker.js]:
```suggestion
  process: async () => {
```

Please remove unused parameters in this file

File: pkg/webui/console/views/admin-packet-broker/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
```
Comment (bafonins) [inline pkg/webui/console/views/admin-packet-broker/index.js]:
```suggestion
// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
```

**Review by bafonins — APPROVED** [review]

File: pkg/web/oauthclient/cookie_state.go [hunk 3]
```diff
@@ -35,13 +35,19 @@ func init() {
...
 
 // StateCookie returns the cookie storing the state of the console.
-func (oc *OAuthClient) StateCookie() *cookie.Cookie {
+func (oc *OAuthClient) StateCookie(tls bool) *cookie.Cookie {
+	sameSite := http.SameSiteNoneMode
+	if !tls {
+		// SameSite=None will be rejected when not using TLS, so we have to
+		// use SameSite=Lax in this case.
+		sameSite = http.SameSiteLaxMode
+	}
```
Comment (htdvisser) [inline pkg/web/oauthclient/cookie_state.go]: I really don't like this. See my comments in https://github.com/TheThingsNetwork/lorawan-stack/pull/4286

**Review by htdvisser — COMMENTED** [review]
Looks like I got a code-owner request because of the stuff that's in https://github.com/TheThingsNetwork/lorawan-stack/pull/4286. Please see my comments there.

File: pkg/web/oauthclient/cookie_state.go (hunk 3 above)
Comment (kschiffer) [reply pkg/web/oauthclient/cookie_state.go]: Sorry this bit got into this PR accidentally.

File: pkg/packetbrokeragent/grpc_pba.go [hunk 4]
```diff
@@ -39,10 +39,15 @@ type pbaServer struct {
 	cpConn *grpc.ClientConn
 }
 
+var errNotEnabled = errors.DefineFailedPrecondition("not_enabled", "Packet Broker is not enabled")
```
Comment (adriansmares) [inline pkg/packetbrokeragent/grpc_pba.go:42]: A noop `pbaServer` would have been cleaner I think, but this is fine as is.

**Review by adriansmares — APPROVED** [review]
Code owned files LGTM. Only have a small comment.

On the design side: maybe add legend below the policy table in order to make it easier for users to know what each sign means ? I'm afraid the hover won't be enough.

File: pkg/packetbrokeragent/grpc_pba.go (hunk 4 above)
Comment (johanstokking) [reply pkg/packetbrokeragent/grpc_pba.go:42]: Yeah I agree, I'll keep it in mind for future PBA work.

**Comment by kschiffer** [comment]
> On the design side: maybe add legend below the policy table in order to make it easier for users to know what each sign means ? I'm afraid the hover won't be enough.

Yeah, the clarity could be improved by that, but I think it's sufficient as it is now also given that this is a feature for admins.

File: cypress/integration/console/packet-broker/routing-policies.spec.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      })
+
+    cy.findByTestId('error-notification').should('not.exist')
+    cy.findByTestId('toast-notification')
+      .should('be.visible')
+      .findByText('Routing policy set')
+      .should('be.visible')
+  })
+
+  it('succeeds unsetting individual per-network routing policy (without default policy)', function () {
```
Comment (bafonins) [inline cypress/integration/console/packet-broker/routing-policies.spec.js:125]:
```suggestion
  it('succeeds unsetting individual per-network routing policy (without default policy)', () => {
```

Comment (kschiffer) [reply cypress/integration/console/packet-broker/routing-policies.spec.js:125]: It needs to use standard functions due to arrow function's lexical binding of `this` (used in `L126` and `L146`).

**Pull request merged**